src/infrastructure/ai/dto/summarizeURL.response.ts

```diff
@@ -0,0 +1,33 @@
+import { SummarizeURLContent } from '@src/infrastructure/ai/types/types';
+
+type SummarizeSuccessType = {
+  success: true;
+  isUserCategory: boolean;
+  response: SummarizeURLContent;
+};
+
+type SummarizeFailType = {
+  success: false;
+  message: string;
+};
+
+type SummarizeResultType = SummarizeSuccessType | SummarizeFailType;
+
+export class SummarizeURLContentDto {
+  success: boolean;
+  isUserCategory?: boolean;
+  response?: SummarizeURLContent;
+  message?: string;
+
+  constructor(data: SummarizeResultType) {
+    // true를 명시하지 않으면 Discriminate Union이 동작 안함
+    if (data.success === true) {
+      this.success = data.success;
+      this.isUserCategory = data.isUserCategory;
+      this.response = data.response;
+    } else {
+      this.success = data.success;
+      this.message = data.message;
+    }
```

Comment:
(/¯◡‿◡)/¯✧·˚ : *✧·˚ : *
중복이니까 이렇게? ㅎㅎ
```suggestion
this.success = data.success;
if (data.success === true) {
      this.isUserCategory = data.isUserCategory;
      this.response = data.response;
    } else {
      this.message = data.message;
```

Reply:
오 이부분 생각 못했었다. 바로 반영할께!

Reply:
해결완료!

[Commit](https://github.com/mash-up-kr/Dorabangs_Node/pull/20/commits/28b961655bcec314d9d5ad279021bbc6cceb41ee)